compiler/typecheck/TcExpr.hs

```diff
@@ -266,6 +266,86 @@ tcExpr e@(HsLamCase x matches) res_ty
               , text "requires"]
     match_ctxt = MC { mc_what = CaseAlt, mc_body = tcBody }
 
+-- For inline bindings, we do essentially the opposite of 'static'
+-- expressions: whereas to typecheck @(static e) :: p a@ we check @e ::
+-- a@, for @(<- e) :: a@ we check @e :: m a@ for some @m@.
+
+tcExpr (HsInlineBind x expr) res_ty
+  = do { res_ty <- expTypeToType res_ty
+{-
+       ; (expr', _lie) <- captureConstraints
+           $ addErrCtxt
+             (hang (text "In the body of an inline binding:") 2 (ppr expr))
+           $ tcPolyExprNC expr ( res_ty
+       ; (co, (_m_ty, expr_ty)) <- matchExpectedAppTy res_ty
+-}
+
+        ; m_ty <- newFlexiTyVarTy (mkFunTy VisArg liftedTypeKind liftedTypeKind)
```

Comment:
`mkFunTy` seems to work here, but I’m not sure if it’s the best way to write the kind `* -> *` since it doesn’t seem to appear elsewhere in the codebase.